easybuild/easyconfigs/p/Python/Python-3.11.5-custom-ctypes.patch

```diff
@@ -0,0 +1,150 @@
++
+ if __version__ != _ctypes_version:
+     raise Exception("Version number mismatch", __version__, _ctypes_version)
+ 
+@@ -349,6 +352,12 @@
+             flags |= _FUNCFLAG_USE_ERRNO
+         if use_last_error:
+             flags |= _FUNCFLAG_USE_LASTERROR
++        # define CDLL instance name as fullpath
++        if _os.name == "posix":
++            if name:
++                fullpath = util.find_library(name)
++                if fullpath is not None:
++                    self._name = fullpath
+         if _sys.platform.startswith("aix"):
+             """When the name contains ".a(" and ends with ")",
+                e.g., "libFOO.a(libFOO.so)" - this is taken to be an
+diff -ruN Python-3.11.5/Lib/ctypes/util.py Python-3.11.5.patched/Lib/ctypes/util.py
+--- Python-3.11.5/Lib/ctypes/util.py	2023-08-24 14:09:18.000000000 +0200
++++ Python-3.11.5.patched/Lib/ctypes/util.py	2025-08-27 17:30:51.896990336 +0200
+@@ -146,8 +146,61 @@
+             # shared objects. See bpo-41976 for more details
+             if not _is_elf(file):
+                 continue
+-            return os.fsdecode(file)
++            file = os.fsdecode(file)
++            # Prevent using CUDA libraries provided on stubs
++            if re.search(r'cuda.*stubs', file, re.IGNORECASE):
++                continue
++            return file
+ 
++    def _findLib_gcc_fullname(name):
++        # Run GCC's linker with the -t (aka --trace) option and examine the
++        # library name it prints out. The GCC command will fail because we
++        # haven't supplied a proper program with main(), but that does not
++        # matter. This function will return the full path of the requested library
++        expr = r'^[^\(\)\s]*%s[^\(\)\s]*' % re.escape(name)
++        c_compiler = shutil.which('gcc')
++        if not c_compiler:
++            c_compiler = shutil.which('cc')
++        if not c_compiler:
++            # No C compiler available, give up
++            return None
++
++        temp = tempfile.NamedTemporaryFile()
++        try:
++            args = [c_compiler, '-Wl,-t', '-o', temp.name, '-l:' + name]
++            # print(args)
++            env = dict(os.environ)
++            env['LC_ALL'] = 'C'
++            env['LANG'] = 'C'
++            try:
++                proc = subprocess.Popen(args,
++                                        stdout=subprocess.PIPE,
++                                        stderr=subprocess.STDOUT,
++                                        env=env)
++            except OSError:  # E.g. bad executable
++                return None
++            with proc:
++                trace = proc.stdout.read().decode()
++        finally:
++            try:
++                temp.close()
++            except FileNotFoundError:
++                # Raised if the file was already removed, which is the normal
++                # behaviour of GCC if linking fails
++                pass
++        res = re.findall(expr, trace, re.MULTILINE)
```

Comment:
If the `re.MULTILINE` is really needed, this needs to be changed in `_findLib_gcc`